pagopa/io-dev-api-server
# [IOCOM-500] Support for cancelled SEND message

⚠️ This PR depends on #294 ⚠️

## Short description
This PR adds support for cancelled PN notification messages, with proper timeline event creation.

## List of changes proposed in this pull request
- `src/features/pn/payloads/messages`: added status property to PN message and proper timeline computation to include both paid events and cancelled status (if needed)
- `src/features/pn/types/notificationStatusHistoryElement`: type for PN timeline elements
- `src/utils/payment`: added rptId computation method from NotificationPaymentInfo instance. Added isPaid method to check when a payment is paid

## How to test
Run the server instance with a template configuration that cancels the notification. Retrieve it using the `third-party-messages/:id` endpoint and check both the `status` property and the related timeline events
